### PR TITLE
Update stylis dependency to 3.x

### PR DESCRIPTION
**Pull request by theshortcut**
Updates to stylis 3.3 and should help address issues like https://github.com/styled-components/styled-components/issues/1218 in the future.

### CODE REVIEW

File: src/test/styles.test.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ describe('with styles', () => {
       ${rule1}
     `
     shallow(<Comp />)
-    expectCSSMatches('.sc-a {} .b { background-color: blue; } .b:hover { -webkit-text-decoration: underline; text-decoration: underline; }')
```
Comment (mxstbr) [inline src/test/styles.test.js]: Any ideas why this doesn't get prefixed anymore? /cc @thysultan

File: src/test/props.test.js [hunk 2]
```diff
@@ -34,6 +34,6 @@ describe('props', () => {
       })};
     `
     shallow(<Comp fg="red" />)
-    expectCSSMatches('.sc-a {} .b { border-width: 0; ; ; }')
```
Comment (mxstbr) [inline src/test/props.test.js]: @wmertens will be very happy to see this fixed! 💯

File: src/test/styles.test.js (hunk 1 above)
Comment (thysultan) [reply src/test/styles.test.js]: Vendor prefixing `text-decoration` was a bug.

Comment (mxstbr) [reply src/test/styles.test.js]: Perfect, thank you for confirming.

**Pull request merged**